cnd/src/swap_digest.rs

```diff
@@ -0,0 +1,170 @@
+use digest::{digest, DigestField, DigestRoot};
+use digest_macro_derive::DigestRootMacro;
+
+use multihash::Multihash;
+
+struct NewType(String);
+
+impl DigestRoot for NewType {
```

Comment:
I reviewed patch by patch initially, now going through 'files changed'.  Please excuse the double posts.

This whole file is testing right?  Can it all go under a conditionally compiled module please i.e., `cfg(test)`.  Perhaps it should all be in the `digest` crate?

Reply:
Excellent point.